app2.py

```diff
@@ -0,0 +1,264 @@
+                }
+
+        r = self.session.put(url, headers=self.session.headers, json=json)
+        return r.status_code
+
+    # 4 
+    def _delete_user(self) -> str:
+        # Notes: Requires Artifactory Pro
+        # Usage: DELETE /api/security/users/{userName}
+        username = input("Enter username to delete: ")
+        if username == "" or username is None:
+            raise ValueError("Missing username")
+        endpoint = f'security/users/{username}'
+        url = self._set_url(endpoint)
+        r = self.session.delete(url, headers=self.session.headers)
+        return r
+
+    # 5
+    def _get_storage_info(self) -> json:
+        # Returns storage summary information regarding binaries, file store and repositories.
+        # Usage: GET /api/storageinfo
+        endpoint = f'storageinfo'
+        url = self._set_url(endpoint)
+        r = self.session.get(url, headers=self.session.headers)
+        return r.json()
+
+    # 6
+    def _create_repository(self) -> requests.status_codes:
+        # Notes: Requires Artifactory Pro
+        # Creates a new repository in Artifactory with the provided configuration. Supported by local, remote, virtual and federated repositories. 
+        # Usage: PUT /api/repositories/{repoKey}
+        repokey = input("Enter repository name: ")
+        if repokey == '' or repokey is None:
+            raise ValueError("repository name cannot be empty")
+
+        json = {
+            'rclass' : 'remote',
+            'url' : f'{self.url}',
+            'externalDependenciesEnabled': False # False (default, Applies to Docker repositories only)
+        }
+        endpoint = f'repositories/{repokey}'
+        url = self._set_url(endpoint)
+        r = self.session.put(url, headers=self.session.headers, json=json)
+        return r.status_code
+
+    # 7
+    # TODO:
+    def _update_repo(self):
+        pass
+
+    # 8
+    def _list_repositories(self) -> json:
+        # Returns a list of minimal repository details for all repositories of the specified type.
+        # Usage: GET /api/repositories[?type=repositoryType (local|remote|virtual|federated|distribution)]|
+        endpoint = 'repositories'
+        url = self._set_url(endpoint)
+        r = self.session.get(url, headers=self.session.headers)
+        return r.json()
+
+    # 9
+    def _change_group(self, group: str) -> str:
+        self.group = group
+
+    # 10
+    # TODO:
+    def _create_group(self):
+        pass
+
+    def _display_menu(self) -> None:
+        menu_options = {
+            1: "Ping",
+            2: "Artifactory Version",
+            3: "Create User",
+            4: "Delete User",
+            5: "Get Storage Info",
+            6: "Create Repository",
+            7: "Update Repository",
+            8: "List Repositories",
+            9: f"Change Group (current group: '{self.group}')",
+            10: "Create Group",
+            11: "Exit",
+            12: "Print Token"
```

Comment:
Maybe use an enum here? for example:
`Class Op(enum):`
`     PING = "Ping"`
`     ARTIFACTORY_VERSION: "Artifactory Version"`

And then 
1: Op.PING
2. Op.ARTIFACTORY_VERSION